test/e2e/framework/gateways.go

```diff
@@ -183,17 +183,23 @@ func (f *Framework) DeleteGateway(cluster ClusterIndex, name string) {
 	}, NoopCheckResult)
 }
 
+func (f *Framework) SaveGatewayNodeForTest(cluster ClusterIndex, gwNode string) {
+	gatewayTest := make(map[int][]string)
+	gatewayTest[int(cluster)] = append(gatewayTest[int(cluster)], gwNode)
+	TestGatewayNodes = gatewayTest
+	By(fmt.Sprintf("Saved GW node for GatewayFailover test - %v", TestGatewayNodes))
```

Comment:
This will recreate the map on each call and erase any previous settings. The map should be created on construction of the `Framework` instance. Then this code becomes:

```suggestion
        f.gatewayNodesToReset[cluster] = append(f.gatewayNodesToReset[cluster], gwNode)
```

Reply:
Done